sdk/storage/storage-queue/src/QueueServiceClient.ts

```diff
@@ -540,4 +441,108 @@ export class QueueServiceClient extends StorageClient {
+    options: ServiceGetPropertiesOptions = {}
+  ): Promise<Models.ServiceGetPropertiesResponse> {
+    const { span, spanOptions } = createSpan(
+      "QueueServiceClient-getProperties",
+      options.spanOptions
+    );
+    try {
+      return this.serviceContext.getProperties({
+        abortSignal: options.abortSignal,
+        spanOptions
+      });
+    } catch (e) {
+      span.setStatus({
+        code: CanonicalCode.UNKNOWN,
+        message: e.message
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Sets properties for a storage account’s Queue service endpoint, including properties
+   * for Storage Analytics, CORS (Cross-Origin Resource Sharing) rules and soft delete settings.
+   * @see https://docs.microsoft.com/en-us/rest/api/storageservices/set-queue-service-properties
+   *
+   * @param {Models.QueueServiceProperties} properties
+   * @param {ServiceGetPropertiesOptions} [options] Options to set properties operation.
+   * @returns {Promise<Models.ServiceSetPropertiesResponse>} Response data for the Set Properties operation.
+   * @memberof QueueServiceClient
+   */
+  public async setProperties(
+    properties: Models.QueueServiceProperties,
+    options: ServiceGetPropertiesOptions = {}
+  ): Promise<Models.ServiceSetPropertiesResponse> {
+    const { span, spanOptions } = createSpan(
+      "QueueServiceClient-setProperties",
+      options.spanOptions
+    );
+    try {
+      return this.serviceContext.setProperties(properties, {
+        abortSignal: options.abortSignal,
+        spanOptions
+      });
+    } catch (e) {
+      span.setStatus({
+        code: CanonicalCode.UNKNOWN,
+        message: e.message
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Retrieves statistics related to replication for the Queue service. It is only
+   * available on the secondary location endpoint when read-access geo-redundant
+   * replication is enabled for the storage account.
+   * @see https://docs.microsoft.com/en-us/rest/api/storageservices/get-queue-service-stats
+   *
+   * @param {ServiceGetStatisticsOptions} [options] Options to get statistics operation.
+   * @returns {Promise<Models.ServiceGetStatisticsResponse>} Response data for get statistics the operation.
+   * @memberof QueueServiceClient
+   */
+  public async getStatistics(
+    options: ServiceGetStatisticsOptions = {}
+  ): Promise<Models.ServiceGetStatisticsResponse> {
+    const { span, spanOptions } = createSpan(
+      "QueueServiceClient-getStatistics",
+      options.spanOptions
+    );
+    try {
+      return this.serviceContext.getStatistics({
+        abortSignal: options.abortSignal,
+        spanOptions
+      });
+    } catch (e) {
+      span.setStatus({
+        code: CanonicalCode.UNKNOWN,
+        message: e.message
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * createQueue and deleteQueue goes here
```

Comment:
>    [](start = 0, length = 3)

is this to be done?

Reply:
This is the only thing that is left in comparison to .NET.
![image](https://user-images.githubusercontent.com/10452642/66880809-548b3000-ef79-11e9-87da-1771f3963ce8.png)


They basically call the getQueueClient() and invoke the create and delete methods on it.

https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Queues/src/QueueServiceClient.cs#L633

Should I do this in this same PR or a followup PR?

Reply:
Yes please, so we have the complete api shape.

Reply:
Added. Done.